working/augmentation-libraries/feature-specification.md

```diff
@@ -721,16 +729,85 @@ It is a compile-time error if:
+*   The augmented constructor is a factory constructor and the augmenting
+    constructor is not or vice versa.
+
+*   The resulting constructor is not valid (has a redirecting initializer and
+    other initializers, multiple `super` initializers, etc).
+
+#### Non-redirecting generative constructors
+
+These are probably the most complex constructor, but also the most common.
+
+A non-redirecting generative constructor marked `augment` may:
+
+*   Add or replace the body of the existing constructor with a new body.
+
+    *   If the augmenting constructor has a body (a block or => body, not just
+        a single ;), then that body replaces the existing constructor body.
+
+    *   In the augmenting constructor's body, an `augmented()` call executes the
+        original constructor's body in the same parameter scope that the
+        augmenting body is executing in. The return type of this call is `void`.
+
+    *   Initializer lists _are not_ re-ran, they have already executed and
+        shouldn't be executed twice.
+
+    *   If a parameter variable is overwritten prior to calling `augmented()`,
+        the augmented body will see the updated value, because the parameter
+        scope is identical.
+
+    *   Local variables from the augmenting scope are not available in the
+        augmented scope.
+
+*   Add initializers to the initializer list.
+
+    *   These are appended to the original constructor's initializers, but before
+        any super initializer (if present).
+
+    *   They are allowed to add a super initializer, if none is present. It must
+        belisted last in the augmenting declaration and will be appended to the end
+        of the initializer list.
+
+    *   It is an error to add redirecting initializers to a non-redirecting
+        generative constructor.
+
+#### Non-redirecting factory constructors
+
+A non-redirecting factory constructor marked `augment` works in the same way as
+a normal function augmentation.
+
+If it has a body, it replaces the body of the augmented constructor, and it may
+invoke the augmented body by calling `augmented(arguments)`.
+
+#### Redirecting generative constructors
+
+A redirecting generative constructor marked `augment` adds its redirecting
+initializer to the augmented constructors initializer list.
+
+This converts it into a redirecting generative constructor, instead of a regular
+generative constructor.
+
+It is a compile-time error if:
+
+*   The augmented constructor has any initializers or a body.
+
+#### Redirecting factory constructors
+
+A redirecting factory constructor marked `augment` adds its constructor call to
+the augmented constructor.
+
+This converts it into a redirecting factory constructor, instead of a regular
+factory constructor.
+
+It is a compile-time error if:
 
-*   The original constructor has a super initializer or redirecting initializer
-    and the augmenting constructor does too.
+*   The augmented constructor has a body.
```

Comment:
(Why? If we treat factory constructors like static methods, just with two ways to specify an implementation, there is no problem in replacing a body with a redirection. 

If you can't agument with `= Foo;`, but can augment with `{ return Foo(args); }`, it's not a distinction, just a distraction.

Reply:
I think it is cleaner to consider redirecting or not uniformly across regular constructors and factory constructors.

Why do we even have redirecting factory constructors at all? Honestly I don't know, but I prefer to keep them separate. We don't have redirecting static methods... so it feels different for sure.